src/app/shared/components/course-form/course-form.component.ts

```diff
@@ -42,11 +68,36 @@ export class CourseFormComponent {
     });
   }
 
+  loadCourseForEdit() {
+    this.buildForm();
+    this.courseService.getCourse(this.courseId).subscribe(course => {
+      this.courseForm.patchValue({
+        title: course.result.title,
+        description: course.result.description,
+        duration: course.result.duration,
+      });
+      const authorsFormArray = this.courseForm.get('authors') as FormArray;
+      const courseAuthorsArray = this.allAuthors.filter(author => {
```

Comment:
This filter looks fuzzy. Why not simplify it to `author => course.result.authors.includes(author.id)`

Reply:
Yes, somehow i overcomplicated it, this looks much better now